### ISSUE TITLE
Support options in external plugins

### ISSUE THREAD

**Issue by azz** [prettier/prettier#3615]
#3433 moved the definition of options to our internal parsers but there's still a bit of work to support external plugins defining new options:

Use cases:

- [ ]  This should include the options contributed by plugins:
  ```bash
  prettier --plugin @prettier/plugin-python --help
  ```

- [ ] CLI should work:
   ```bash
   prettier --plugin my-cool-prettier-plugin --my-custom-option
   ```

- [ ]  Remove unknown option warning for options contributed by external plugins.

   ```js
   prettier.format('def foo(): pass', {
     parser: 'python',
     plugins: ['@prettier/plugin-python'],
     pythonVersion: '3.1.4'
   })
  ● Validation Warning:

    Unknown option "pythonVersion" with value "3.1.4" was found.
    This is probably a typing mistake. Fixing it will remove this message.
  ```

**Comment by taion** [comment]
I don't believe #3433 did that. I believe it just exposed (via some code duplication) plugin options in `getSupportInfo()`.

At the moment, all options other than the ones in `cli/constant` get dropped: https://github.com/prettier/prettier/blob/6c0dd745185bfd20bb3b554dd591b8ef0b480748/src/cli/util.js#L628-L642. In fact this happens before plugins are even loaded.

Unless I'm missing something, supporting per-plugin options would require a revamp of option parsing here. You'd need to first load plugins, then load their options, _then_ parse the rest of the config.

**Comment by azz** [comment]
It will just need to key into `options.js`'s `normalize`. Shouldn't be too much work.

**Comment by taion** [comment]
Yeah, something like that. There are a few different code paths here depending on which entry point you use. Also a bit harder for CLI stuff if relevant. Nothing too bad though.

Though it is worth thinking about how to configure e.g. trailing commas for Python. Something other than `--trailing-comma` would be most straightforward, but slightly messy.

**Comment by duailibe** [comment]
@ikatyang has already planned a follow-up PR to deduplicate work with `normalize`. #3433 was just the start in that direction to unblock some other stuff (like #3573)

**Comment by ikatyang** [comment]
use supportOptions to generate CLI options (internal plugins) --> #3622
